Added integration with sbt-header like in akka-http

kelebra opened the pull request:
This PR introduces integration with [sbt-headers](https://github.com/sbt/sbt-header) discussed in [this PR](https://github.com/akka/akka/pull/24568) based on implementation in [another PR from akka-http](https://github.com/akka/akka-http/pull/1738) in order to automate population of copyright header in source code. 

Main points of changes: `project/plugins.sbt`, `project/CopyrightHeader.scala`, `build.sbt`. All other files are changed due to execution of `headerCreate` sbt command.

I am no sbt expert however this seemed like a low hanging fruit which could improve contribution experience for others especially the ones who contribute to akka for the first time.

kelebra committed:
Added integration with sbt-header like in akka-http

kelebra committed:
Merge branch 'master' into sbt-header-integration

# Conflicts:
#	akka-remote/src/main/java/akka/remote/artery/aeron/AeronErrorLog.java
#	akka-testkit-typed/src/main/scala/akka/testkit/typed/javadsl/ExplicitlyTriggeredScheduler.scala

kelebra committed:
Pushing more results of headerCreate based on merge

johanandren commented:
OK TO TEST

ktoso commented:
Excellent, I'm all for merging it -- perhaps after we merge things for today's release.

patriknw commented on akka-actor-typed/src/main/scala/akka/actor/typed/javadsl/TimerScheduler.scala:
```diff
@@ -1,6 +1,7 @@
 /**
  * Copyright (C) 2017-2018 Lightbend Inc. <https://www.lightbend.com>
  */
+
```
do we have to add these newlines?

kelebra replied:
In my opinion that looks consistent with `akka-http`

raboof replied:
They're also there in `akka-http` indeed, but they do look a bit strange to me as well. Do we want them?

patriknw commented on akka-actor-typed/src/main/scala/akka/actor/typed/package.scala:
```diff
@@ -1,3 +1,7 @@
+/*
+ * Copyright (C) 2018 Lightbend Inc. <https://www.lightbend.com>
+ */
```
good one

patriknw commented on akka-actor/src/main/java/akka/dispatch/forkjoin/package-info.java:
```diff
@@ -4,7 +4,6 @@
  * http://creativecommons.org/publicdomain/zero/1.0/
  */
```
Good that it doesn't change that

patriknw replied:
Do we have any other headers that have both Lightbend and some other copyright notice? When we copy code from elsewhere (when the license permits) we preserve the original copyright notice.

ktoso replied:
akka-protobuf

Philippus replied:
akka-persistence

patriknw reviewed (commented):
great, finally a smart pattern that preserves the original years

kelebra committed:
Merge branch 'master' into sbt-header-integration

kelebra committed:
Added header population on compilation

kelebra commented:
Things that changed after last commits:

- In sync with master.
- Added header generation on compilation in order to simplify regular development.
- Added tests into scope. Actually more files are formatted now especially in `akka-testkit-typed` and other new modules.

kelebra committed:
Fixed duplication in formatting

kelebra committed:
Execution of test:compile. Fixing code formatting and copyright.

ktoso commented on akka-actor-tests/src/test/java/akka/actor/ActorSystemTest.java:
```diff
@@ -1,4 +1,8 @@
+
+/**
   * Copyright (C) 2009-2018 Lightbend Inc. <https://www.lightbend.com>
   */ 
```
looks wrong?

kelebra replied:
Probably something due to extra space in the beginning of each line. I would actually expect MiMa to format it correctly or complain about it. Anyway this could be also fixed with a bit of enhancing of detection regexp.

kelebra committed:
LF endings

kelebra committed:
Fixing test case which got impacted after adding linebreak in copyright

kelebra commented:
Looks like all done. Sorry for mess, looks like I actually broke build somehow accidentally changing line endings from `LF` to `CRLF` (issues with local setup).

Please let me know if there is anything that should be done on this PR.

Philippus commented on akka-actor-tests/src/test/java/akka/actor/JavaAPITestActor.java:
```diff
@@ -1,3 +1,7 @@
+/*
+ * Copyright (C) 2018 Lightbend Inc. <https://www.lightbend.com>
```
should this be `2011-2018` in this case? Or do we not care?

raboof replied:
I think the sbt-header configuration just adding `2018` is OK. We could consider manually updating files we know are older, but I'm OK with the header like this as well. AFAIK the year in a copyright header is mostly there for informational purposes, not really legal ones.

raboof reviewed (commented):
We might want to remove the extra newlines and look into the problem @ktoso noticed, but in general super nice PR! Looking forward to having it in :)

kelebra commented:
Thanks for further steps, I will revisit this PR tomorrow. Probably line endings should not be an issue and will start from there.

kelebra commented:
Actually, looks like it is an issue 😢 See [this rejected PR for more info](https://github.com/sbt/sbt-header/pull/117), looks like play and others are okay with it.

kelebra commented:
So unfortunately no way around it so far. Root cause of these extra line ending is [here](https://github.com/sbt/sbt-header/blob/master/src/main/scala/de/heikoseeberger/sbtheader/CommentStyle.scala#L31). Sorry guys but it is pretty much closed by `final`.
